### PR TITLE
[EV-1234] [EV-1268] add paginated UTXOs function

### PR DESCRIPTION
**Pull request by Cofresi**
**Issue being fixed or implemented**
add new insight api endpoint
/addrs/[:addrs]/utxopage

This will be used as for the DAPI endpoint getUTXO to speed up requests for UTXOs for addresses with hundreds of thousands of unspent outputs like coinbase accumulating faucet addresses. The default pagination was set to 0 (from) and 1000 (to) respectively, meaning that there is only a pagination with a corresponding capped result set if there are more than 1000 outputs for the addresses to be queried. 1000 has been chosen because it seems that problematic time lags are happening only with outputs numbering significantly above this threshold.

new parameters for pagination:

from
to
fromHeight
toHeight

combinations of both range types are possible. Keep in mind that the order of the result list is always most recent UTXO on top.

**What was done**
added new insight api endpoint

**What has been tested**
unit tests and on testnet insight 
http://195.141.143.49:3001/insight-api/addrs/ybFPBD7hm9KVd2Dubj97K5mw2ymR8gWJre/utxopage
http://195.141.143.49:3001/insight-api/addrs/ybFPBD7hm9KVd2Dubj97K5mw2ymR8gWJre/utxopage?fromHeight=47412
http://195.141.143.49:3001/insight-api/addrs/ybFPBD7hm9KVd2Dubj97K5mw2ymR8gWJre/utxopage?fromHeight=47412&toHeight=47420
http://195.141.143.49:3001/insight-api/addrs/ybFPBD7hm9KVd2Dubj97K5mw2ymR8gWJre/utxopage?from=2&to=12
http://195.141.143.49:3001/insight-api/addrs/ybFPBD7hm9KVd2Dubj97K5mw2ymR8gWJre/utxopage?from=1002
http://195.141.143.49:3001/insight-api/addrs/ybFPBD7hm9KVd2Dubj97K5mw2ymR8gWJre/utxopage?from=2&to=12&fromHeight=47412&toHeight=47420

**What needs to be tested**
n/a

**Comments**
needs to be merged after https://github.com/dashevo/dashcore-node/pull/54

### CODE REVIEW

File: lib/addresses.js [hunk 1]
```diff
@@ -226,6 +226,38 @@ AddressController.prototype.transformAddressHistoryForMultiTxs = function(txinfo
...
 
+AddressController.prototype.paginatedUtxo = function(req, res, next) {
+  var self = this;
+
+  var options = {
+    from: parseInt(req.query.from) || parseInt(req.body.from) || 0
+  };
+
+  options.to = parseInt(req.query.to) || parseInt(req.body.to) || parseInt(options.from) + 1000;
+  if (options.from >= options.to) {
```
Comment (abvgedeika) [inline lib/addresses.js]: why we can't getUtxo when 'from' equals 'to'?

Comment (Cofresi) [reply lib/addresses.js]: Good catch! That should really be allowed :). Will update

**Review by antouhou — DISMISSED** [review]
utACK

**Review by nmarley — COMMENTED** [review]
I noticed you've added both a `get` and a `post` route -- is this desired, or just follows from the above convention for a few routes? I also see that not all routes match both get and post.

Another concern is the removal of `package-lock.json` in this and the aforementioned dashcore-node PR. What is the reason for this? I thought it was recommended to keep these checked in b/c these lock specific versions of dependencies.

**Comment by Cofresi** [comment]
> I noticed you've added both a `get` and a `post` route -- is this desired, or just follows from the above convention for a few routes? I also see that not all routes match both get and post.
> 

I just copied the convention from the `/utxo` route.

> Another concern is the removal of `package-lock.json` in this and the aforementioned dashcore-node PR. What is the reason for this? I thought it was recommended to keep these checked in b/c these lock specific versions of dependencies.

whoops, no idea why this was missing. Added it back

**Review by antouhou — APPROVED** [review]
utACK. We can merge it if nobody else has any objections

**Comment by antouhou** [comment]
Hmm, it looks like only @nmarley can approve this one, since he is the code owner

**Review by nmarley — APPROVED** [review]
utACK

**Pull request merged**